docs/2_1_common_api.adoc

```diff
@@ -1047,6 +1049,64 @@ _After each such call, the elements of the resulting directional derivative vect
 
 _More details and implementational notes are available from <<ABL12>>.]_
 
+===== Adjoint derivatives
+
+If the capability attribute `providesAdjointDerivative = true`, <<fmi3GetAdjointDerivative>> computes the adjoint deriatives of an FMU.
+
+_[Adjoint derivatives are beneficial in several contexts_
+
+* _in Artifical Intelligence (AI) frameworks the adjoint derivatives are called "VJP"s meaning vector-gradient-products. There adjoint derivatives are used in the backpropagation process to do gradient-based optimization of parameters using reverse mode automatic differentiation (AD), see, e.g., <<BPRS15>>._
+* _in pararameter estimation (see <<BKF17>>)_
+
+_Typically, reverse mode automatic differentiatian (AD) is much more efficient for these use cases than forward mode AD, as explained in the cited references._
+
+_If one would like to construct the full Jacibian matrix, one can use either <<fmi3GetDirectionalDerivative>> (to column-wise construct it) or <<fmi3GetAdjointDerivative>> (to row-wise construct it, possibly improved with coloring methods as mentioned above)._
```

Comment:
_Add_: However, for computing the full Jacobian matrix the column-wise construct is generally more efficient.
(I know that we shouldn't discuss implementation issues etc, but if we say that there are two ways of computing something and we know that one will be more efficient we should say that and not lead users down the wrong path.)

Reply:
That depends on the sparsity pattern of the Jacobian.

Reply:
@HansOlsson:
> _Add_: However, for computing the full Jacobian matrix the column-wise construct is generally more efficient.

Is there an explanation or reference for this?

This seems also very much to depend on the dimenstions:
https://arxiv.org/pdf/1811.05031.pdf provides examples where either one or the other has compuational advantages depending, e.g. on the dimensions n and m of the m*n Jacobian , e.g. 
![grafik](https://user-images.githubusercontent.com/16363372/74464070-2e2bc180-4e93-11ea-8465-1cfc25a4f900.png)

Reply:
For R^n -> R^m, forward mode is O(n) and reverse mode is O(m) with a larger constant. So if n is approximately m, forward mode will be better. But if m<<n, then reverse mode is better. When you do matrix coloring of a sparse matrix, it's the same idea except the number of colors for the column vs the row space. So for the standard case of R^n -> R^n Jacobians like in a stiff ODE solver forward mode is almost always better, in general it's much more complex.